src/TurbulenceClosures/turbulence_closure_implementations/Smagorinskys/scale_invariant_operators.jl

```diff
@@ -41,11 +41,19 @@ using Oceananigans.Operators: volume
 ##### Filtering
 #####
 
-@inline ℱ₂ᶠᶜᶜ(i, j, k, grid, f, args...) = ℑxyzᶠᶜᶜ(i, j, k, grid, ℑxyzᶜᶠᶠ, f, args...)
-@inline ℱ₂ᶜᶠᶜ(i, j, k, grid, f, args...) = ℑxyzᶜᶠᶜ(i, j, k, grid, ℑxyzᶠᶜᶠ, f, args...)
-@inline ℱ₂ᶜᶜᶠ(i, j, k, grid, f, args...) = ℑxyzᶜᶜᶠ(i, j, k, grid, ℑxyzᶠᶠᶜ, f, args...)
-@inline ℱ₂ᶜᶜᶜ(i, j, k, grid, f, args...) = ℑxyzᶜᶜᶜ(i, j, k, grid, ℑxyzᶠᶠᶠ, f, args...)
-
+# @inline ℱ₂ᶠᶜᶜ(i, j, k, grid, f, args...) = ℑxyzᶠᶜᶜ(i, j, k, grid, ℑxyzᶜᶠᶠ, f, args...)
+# @inline ℱ₂ᶜᶠᶜ(i, j, k, grid, f, args...) = ℑxyzᶜᶠᶜ(i, j, k, grid, ℑxyzᶠᶜᶠ, f, args...)
+# @inline ℱ₂ᶜᶜᶠ(i, j, k, grid, f, args...) = ℑxyzᶜᶜᶠ(i, j, k, grid, ℑxyzᶠᶠᶜ, f, args...)
+# @inline ℱ₂ᶜᶜᶜ(i, j, k, grid, f, args...) = ℑxyzᶜᶜᶜ(i, j, k, grid, ℑxyzᶠᶠᶠ, f, args...)
+
+@inline ℱ₂ᶠᶜᶜ(i, j, k, grid, u::AbstractArray) = @inbounds (6 * u[i, j, k] +
+                                                            u[i+1, j, k] + u[i-1, j, k] +
+                                                            u[i, j+1, k] + u[i, j-1, k] +
+                                                            u[i, j, k+1] + u[i, j, k-1]) / 12
+const ℱ₂ᶜᶠᶜ = ℱ₂ᶠᶜᶜ
+const ℱ₂ᶜᶜᶠ = ℱ₂ᶠᶜᶜ
+const ℱ₂ᶜᶜᶜ = ℱ₂ᶠᶜᶜ
```

Comment:
@tomchor can you take a look at this change?

I was looking at these operators and I realized that the 4 operators are identical (at least in the case that the reconstruction operators do not depend on grid topology which they don't now). Therefore we only need one filtering function. Also, we only need it for arrays (not functions). I'd like to see how this plays out with compile time but I suspect it will help, since we are compiling just 1 function instead of... 4 * 6 = 24.